src/player.js

```diff
@@ -3,38 +3,88 @@ class Character {
         this.name = name;
         this.hp = 100;
         this._speed = 100;
+        this._isMoving = false;
+        this._destination = {
+            x: Constants.PLAYER_SPAWN_X,
+            y: Constants.PLAYER_SPAWN_Y
+        };
+        this._prevPosition = null;  // initialized with Sprite
+        this._blocked = null;  // initialized with Sprite
     }
 
     attachSprite(sprite) {
         this.sprite = sprite;
         // Normalize and scale the velocity so that player can't move faster along a diagonal
         this.sprite.body.velocity.normalize().scale(this._speed);
         this.sprite.setCollideWorldBounds(true);
+        this.sprite.body.setSize(16, 8);
+        this.sprite.body.setOffset(8, 24);
+        this._prevPosition = { x: this.sprite.x, y: this.sprite.y };
+        this._blocked = this.sprite.body.blocked;
     }
 
     attachController(controller) {
         this.controller = controller;
     }
 
     update() {
-        // Stop any previous movement from the last frame
-        this.sprite.body.setVelocity(0);
-        // Horizontal movement
-        if (this.controller.left.isDown) {
-            this.sprite.body.setVelocityX(-this._speed);
-            this.sprite.anims.play('playerLeft', true);
-        } else if (this.controller.right.isDown) {
-            this.sprite.body.setVelocityX(this._speed);
-            this.sprite.anims.play('playerRight', true);
-        } else if (this.controller.up.isDown) {
-            this.sprite.body.setVelocityY(-this._speed);
-            this.sprite.anims.play('playerUp', true);
-        } else if (this.controller.down.isDown) {
-            this.sprite.body.setVelocityY(this._speed);
-            this.sprite.anims.play('playerDown', true);
-        } else {
-            this.sprite.anims.stopOnRepeat();
+        this._handleMovement();
+    }
+
+    isBlocked() {
+        return this._blocked.down || this._blocked.up || this._blocked.left || this._blocked.right;
+    }
+
+    _moveWithStep(key, stepSizeX, stepSizeY) {
```

Comment:
What means underscore before methods names?? Do we need them?

Reply:
_ before a function means it's "private" or in other words: not public interface or in other words:
avoid changing this functions code :)

Reply:
ok, i see. We can follow this convection.